webClient/src/app/editor/code-editor/monaco/monaco.service.ts

```diff
@@ -307,72 +307,81 @@ export class MonacoService {
-          }
-          });
+        let activeDirectory = '';
+        if (fileDirectory) {
+          activeDirectory = fileDirectory;
         }
-        /* The file was never tagged, so we should
-         * ask the user if they would like to tag
-         * it.
-         */
-        else {
-          /* Open up a dialog asking if the user
-           * wants to tag their file. Again,
-           * we are checking if ISO-8859-1 is
-           * an option.
-           */
-          let saveRef = this.dialog.open(TagComponent, {
-            width: '500px',
-            data: { canBeISO: x,
-                    fileName: fileContext.model.fileName }
-          });
-          saveRef.afterClosed().subscribe(result => {
-          if (result) {
-            this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
-          }
-          });
+        let saveRef = this.dialog.open(SaveToComponent, {
+          width: '500px',
+          data: { canBeISO: x, 
+            fileName: fileContext.model.fileName,
+            fileDirectory: activeDirectory }
+        });
+        saveRef.afterClosed().subscribe(result => {
+        if (result) {
+          this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
         }
+        });
+      }
+
+      /* If the file is not new, and the encoding 
+       * has already been set inside of USS via
+       * chtag.
+       */
+      else
+      {
+        this.editorControl.getFileMetadata(fileContext.model.path + '/' + fileContext.model.name).subscribe(r => {
+          fileContext.model.encoding = r.ccsid;
+          if (!fileContext.temp && 
+            fileContext.model.encoding != undefined &&
+            fileContext.model.encoding != null && 
+            fileContext.model.encoding != 0
+            ) {
+            this.editorControl.saveBuffer(fileContext, null).subscribe(() => obs.next());
+          }
+          /* The file was never tagged, so we should
+          * ask the user if they would like to tag
+          * it.
+          */
+          else {
+            /* Issue a presave check to see if the
+            * file can be saved as ISO-8859-1,
+            * perhaps this should be done in real
+            * time as an enhancement.
+            */
+            let x = this.preSaveCheck(fileContext);
+            /* Open up a dialog asking if the user
+            * wants to tag their file. Again,
+            * we are checking if ISO-8859-1 is
+            * an option.
+            */
+            let saveRef = this.dialog.open(TagComponent, {
+              width: '500px',
+              data: { canBeISO: x,
+                      fileName: fileContext.model.fileName }
+            });
+            saveRef.afterClosed().subscribe(result => {
+              if (result) {
+                this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next());
+              }
+            });
+          }
```

Comment:
This uses a lot of duplicate code with the lines 315-340 so probably plop into a method with optional arg 'fileDirectory' that if exists, goes into 'data' object

or

use short-hand JS syntax to conditionally add properties into 'data' object depending on if they exist
i.e.
```
data = {
    ...condition && { prop: value },
};
```

Reply:
done